mps_youtube/player.py

```diff
@@ -138,18 +146,24 @@ def _launch_player(self):
         """ Launch player application. """
         pass
 
-    def send_metadata_mpris(self):
+    def generate_metadata(self):
         metadata = util._get_metadata(self.song.title)
 
-        if metadata is None:
-            arturl = "https://i.ytimg.com/vi/%s/default.jpg" % self.song.ytid
-            metadata = (self.song.ytid, self.song.title, self.song.length,
-                        arturl, [''], '')
-        else:
-            arturl = metadata['album_art_url']
-            metadata = (self.song.ytid, metadata['track_title'],
-                        self.song.length, arturl,
-                        [metadata['artist']], metadata['album'])
+        arturl = "https://i.ytimg.com/vi/%s/default.jpg" % self.song.ytid
+        ret_meta = OrderedDict(ytid = self.song.ytid, title=self.song.title,
+                               length=self.song.length, arturl=arturl, artists=[''],
+                               album='')
+
+        if metadata is not None:
+            ret_meta['title'] = metadata['track_title']
+            ret_meta['arturl'] = metadata['album_art_url']
+            ret_meta['artists'] = [metadata['artist']]
+            ret_meta['album'] = metadata['album']
+
+        return ret_meta
+
+    def send_metadata_mpris(self):
+        metadata = tuple((v for k,v in self.generate_metadata().items()))
```

Comment:
You can use the `values()` method here: `metadata = tuple(self.generate_metadata().values())`

As a side note, you can leave off the extra set of parentheses with generator comprehensions like this: `tuple(v for k,v in self.generate_metadata().items())`. But with `.values()` you don't even need the generator comprehension.

Reply:
Although on second thought, can't you send the dict to the mpris process? Or a `collections.namedtuple` might be neater. It's probably best to use the same type to store this data in both places instead of converting to another format.